tests/unit/backend/models/form.server.model.spec.ts

```diff
@@ -879,104 +950,116 @@ describe('Form Model', () => {
-        populatedAdmin._id,
-        populatedAdmin.email,
-      )
+      it('should return array of forms user is permitted to view', async () => {
+        // Arrange
+        // Add additional user.
+        const differentUserId = new ObjectId()
+        const diffPreload = await dbHandler.insertFormCollectionReqs({
+          userId: differentUserId,
+          mailName: 'something-else',
+          mailDomain: MOCK_ADMIN_DOMAIN,
+        })
+        const diffPopulatedAdmin = merge(diffPreload.user, {
+          agency: diffPreload.agency,
+        })
+        // Populate multiple forms with different permissions.
+        // Is admin.
+        const userOwnedForm = await Form.create(MOCK_EMAIL_FORM_PARAMS)
+        // Has write permissions.
+        const userWritePermissionForm = await Form.create({
+          ...MOCK_ENCRYPTED_FORM_PARAMS,
+          admin: diffPopulatedAdmin._id,
+          permissionList: [{ email: populatedAdmin.email, write: true }],
+        })
+        // Has read permissions.
+        const userReadPermissionForm = await Form.create({
+          ...MOCK_ENCRYPTED_FORM_PARAMS,
+          admin: diffPopulatedAdmin._id,
+          // Only read permissions, no write permission.
+          permissionList: [{ email: populatedAdmin.email, write: false }],
+        })
+        // Should not be fetched since form is archived.
+        await Form.create({
+          ...MOCK_ENCRYPTED_FORM_PARAMS,
+          status: Status.Archived,
+        })
+        // Should not be fetched (not collab or admin).
+        await Form.create({
+          ...MOCK_ENCRYPTED_FORM_PARAMS,
+          admin: differentUserId,
+          // currentUser does not have permissions.
+        })
 
-      // Assert
-      // Coerce into expected shape.
-      const keysToPick = [
-        '_id',
-        'title',
-        'lastModified',
-        'status',
-        'responseMode',
-      ]
-      const expected = orderBy(
-        [
-          // Should return form with admin themselves.
-          merge(pick(userOwnedForm.toObject(), keysToPick), {
-            admin: populatedAdmin.toObject(),
-          }),
-          // Should return form where admin has write permission.
-          merge(pick(userWritePermissionForm.toObject(), keysToPick), {
-            admin: diffPopulatedAdmin.toObject(),
-          }),
-          // Should return form where admin has read permission.
-          merge(pick(userReadPermissionForm.toObject(), keysToPick), {
-            admin: diffPopulatedAdmin.toObject(),
-          }),
-        ],
-        'lastModified',
-        'desc',
-      )
-      // Should return list containing only three forms:
-      // (admin, read perms, write perms),
-      // even though there are 5 forms in the collection.
-      await expect(Form.countDocuments()).resolves.toEqual(5)
-      expect(actual.length).toEqual(3)
-      expect(actual).toEqual(expected)
+        // Act
+        const actual = await Form.getDashboardForms(
```

Comment:
nit: something like `Form.getByUserId` could be a better name, since the service function shouldn't be coupled with the UI.

Reply:
good point, will update

Reply:
Renamed to `Form.getAllByUserIdOrEmail`

Reply:
Rerenaming it to `Form.getMetaByUserIdOrEmail` since it's actually just retrieving a subset